scripts/lua-entities-util.js

```diff
@@ -0,0 +1,353 @@
+        if (foundList[n]) {
+          return false;
+        }
+        foundList[n] = true;
+        return true;
+      });
+      cb(null, result);
+    });
+
+    lines.forEach(function (line) {
+      parseFile(line, function (err, kvData) {
+        if (err) {
+          return done(err);
+        }
+        var itemList = getItemsFromKV(kvData);
+        result = result.concat(itemList);
+        var luaPathList = getLuaPathsFromKV(kvData);
+
+        var luaPathDone = after(luaPathList.length, done);
+        luaPathList.forEach(function (luaPath) {
+          findLinkLuaModifiersInFile(luaPath, function (err, modifiers) {
+            if (err) {
+              return luaPathDone(err);
+            }
+            result = result.concat(modifiers);
+            luaPathDone();
+          });
+        });
+      });
+    });
+  });
+}
+
+function getUnitsFromKV (data) {
+  return Object.keys(data.DOTAUnits).filter(n => n !== 'values');
+}
+
+function getItemsFromKV (data) {
+  return Object.keys(data.DOTAItems).filter(function (name) {
+    if (name === 'values') {
+      return false;
+    }
+    return !(!data.DOTAItems[name].values.BaseClass || data.DOTAItems[name].values.BaseClass === name);
+  });
+}
+
+function getAbilitiesFromKV (data) {
+  return Object.keys(data.DOTAAbilities).filter(n => n !== 'values');
+}
+
+function getLuaPathsFromKV (data) {
+  return getItemsFromKV(data).map(function (item) {
+    switch (data.DOTAItems[item].values.BaseClass) {
+      case 'item_datadriven':
+        // probably nothing?
+        // console.log(Object.keys(data.DOTAItems[item]));
+        break;
+      case 'item_lua':
+        if (data.DOTAItems[item].values.ScriptFile.endsWith('.lua')) {
+          return data.DOTAItems[item].values.ScriptFile;
+        } else {
+          return data.DOTAItems[item].values.ScriptFile + '.lua';
+        }
+    }
+    return [];
+  })
+  .reduce(function (memo, val) {
+    return memo.concat(val);
+  }, []);
+}
+
+var hiddenModifiers = {};
+function isModifierHidden (modPair, cb) {
+  if (hiddenModifiers[modPair[0]] !== undefined) {
+    return cb(null, hiddenModifiers[modPair[0]]);
+  }
+  var functionString = ['function ', modPair[0], ':IsHidden()'].join('');
+  fs.readFile(path.join(vscriptsDir, modPair[1]), {
+    encoding: 'utf8'
+  }, function (err, data) {
+    if (err) {
+      return cb(err);
+    }
+    var foundLine = false;
+    var result = false;
+    data.split('\n')
+      .forEach(function (line) {
+        line = line.trim();
+        if (foundLine) {
+          result = (line === 'return true');
+          foundLine = false;
+          return;
+        }
+        if (line === functionString) {
+          foundLine = true;
+        }
+      });
+    // store result
+    hiddenModifiers[modPair[0]] = result;
+
+    cb(null, result);
+  });
+}
+
+function findLinkLuaModifiersInFile (script, cb, fullPathProvided = false, failOnNonExistentPath = false) {
```

Comment:
Most of this file was moved from find-tooltips.js. I modified this function a bit though - it allows for reporting errors and detects Link instrouctions preceeded by whitespace.